### PR TITLE
rpm: immutable-object-cache related changes

### PR DESCRIPTION
**Pull request by tchaikov**
* use "%package immutable-object-cache" not "%package
  ceph-immutable-object-cache" for naming subpackage of
  "ceph-immutable-object-cache". this would result in package name of
  "ceph-immutable-object-cache". see
  http://ftp.rpm.org/max-rpm/s1-rpm-subpack-spec-file-changes.html
* remove "libradospp1" from immutable-object-cache's runtime
  dependencies. this package was removed in 65c8733b .

Signed-off-by: Kefu Chai <kchai@redhat.com>

- [ ] References tracker ticket
- [ ] Updates documentation if necessary
- [ ] Includes tests for new functionality or reproducer for bug

### CODE REVIEW

**Comment by tchaikov** [comment]
cc @zhouyuan

File: ceph.spec.in [hunk 1]
```diff
@@ -584,14 +584,13 @@ Requires:	librbd1 = %{_epoch_prefix}%{version}-%{release}
 Daemon for mirroring RBD images between Ceph clusters, streaming
 changes asynchronously.
 
-%package -n ceph-immutable-object-cache
```
Comment (runsisi) [inline ceph.spec.in]: should be `%package immutable-object-cache` ?

**Comment by zhouyuan** [comment]
@tchaikov thanks a lot for the fix!
jenkins retest this please

File: ceph.spec.in (hunk 1 above)
Comment (tserong) [reply ceph.spec.in]: If the package name needs to be ceph-immutable-object-cache, then it should be `%package immutable-object-cache` (i.e. remove `-n` and the `ceph-` prefix, and rpm will automatically add the `ceph-` prefix).

Comment (tchaikov) [reply ceph.spec.in]: @runsisi agreed. after reading @tserong 's comment and rereading the rpm doc referenced in the commit message, i realized i misread the document. so instead, i am removing "-n" and "ceph-" prefix. and split the commit into two:

1. remove `-n` and the `ceph-` prefix, as put by Tim
1. remove libradospp1 runtime dependency.

File: ceph.spec.in [hunk 2]
```diff
@@ -586,7 +586,6 @@ Summary:	Ceph daemon for immutable object cache
 Group:		System/Filesystems
 %endif
 Requires:	ceph-common = %{_epoch_prefix}%{version}-%{release}
-Requires:	libradospp1 = %{_epoch_prefix}%{version}-%{release}
```
Comment (zhouyuan) [inline ceph.spec.in]: @tchaikov sorry for this mess.. 
The daemon needs to read rados objects from RADOS and store them on a local device.  if i read https://github.com/ceph/ceph/commit/65c8733b56d2392ec70d6bf4db994697a4828287 correctly, we still need this dependency here but need to change it to `librados2`?
`Requires:	librados2 = %{_epoch_prefix}%{version}-%{release}`

Comment (tchaikov) [reply ceph.spec.in]: s/ceph-common/librados2/

updated and repushed.

**Comment by tchaikov** [comment]
changelog

- s/ceph-common/librados2/ in immutable-object-cache's dependencies.

**Comment by dillaman** [comment]
https://shaman.ceph.com/repos/ceph/wip-jd-testing/ef6064e3c25d42d3400514fb982c148d17a968f7/

**Comment by tchaikov** [comment]
Jason, thanks for testing it!

**Pull request merged**